### PR TITLE
LIME-468 - Activity History Score logging re-work

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

Updated activity history score logging to reflect updated field location in experian response

### CODE REVIEW

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 1]
```diff
@@ -103,39 +102,55 @@ private FraudCheckResult mapResponse(
...
+                                    .equals("IDandLocDataAtCL_StartDateOldestPrim")) {
+                        Integer dateValue = dataCountObject.getValue();
+
+                        IDandLocDataAtCL_StartDateOldestPrim = dateValue.toString();
+
+                        calculateRecordAge(
+                                IDandLocDataAtCL_StartDateOldestPrim,
+                                "IDandLocDataAtCL_StartDateOldestPrim");
                     }
+                    if (dataCountObjectNotNull
```
Comment (ElliotMurphyGDS) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:126]: I can probably extract this into a method to reduce duplicated code

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 2]
```diff
@@ -103,39 +102,55 @@ private FraudCheckResult mapResponse(
...
-                        calculateOldestDataRecordDate(
-                                authConsumer.getLocDataOnlyAtCLoc().getStartDateOldestPrim(),
-                                "LocDataOnlyAtCLoc_StartDate0ldestPrim");
+                List<DataCount> dataCounts = decisionElement.getDataCounts();
+
+                String IDandLocDataAtCL_StartDateOldestPrim = null;
+                String IDandLocDataAtCL_StartDate0ldestSec = null;
+                String LocDataOnlyAtCLoc_StartDate0ldestPrim = null;
+
+                for (DataCount dataCountObject : dataCounts) {
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:111]: Can we add a unit test where data counts is empty list and null just to make sure its not entering this for in those scenarios

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java [hunk 3]
```diff
@@ -103,39 +102,58 @@ private FraudCheckResult mapResponse(
...
-                        calculateOldestDataRecordDate(
-                                authConsumer.getLocDataOnlyAtCLoc().getStartDateOldestPrim(),
-                                "LocDataOnlyAtCLoc_StartDate0ldestPrim");
+                List<DataCount> dataCounts = decisionElement.getDataCounts();
+
+                String IDandLocDataAtCL_StartDateOldestPrim = null;
+                String IDandLocDataAtCL_StartDate0ldestSec = null;
+                String LocDataOnlyAtCLoc_StartDate0ldestPrim = null;
+
+                if (null != dataCounts) {
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/IdentityVerificationResponseMapper.java:111]: Can we also add a test for where we have all 3 values and one with an invalid date and one with a date in the future.

I think we might need a try catch round the content of calculateRecordAge as we never want to fail if something goes wrong here we want to ignore it and carry on

**Review by gds-jbentham — APPROVED** [review]
